src/base/bittorrent/peerinfo.cpp

```diff
@@ -198,7 +198,16 @@ QString PeerInfo::I2PAddress() const
 
 QString PeerInfo::client() const
 {
-    return QString::fromStdString(m_nativeInfo.client);
+    QString client = QString::fromStdString(m_nativeInfo.client);
+
+    for(QChar& c : client)
+    {
+        // replace non-printable characters with spaces
+        if (!c.isPrint())
+            c = u' ';
```

Comment:
Maybe use `QChar::ReplacementCharacter`?
https://en.wikipedia.org/wiki/Specials_(Unicode_block)#Replacement_character
IMO it is ugly but it is better than mixing up unprintable chars with a true space.

Reply:
Or `☐`, `⍰` as mentioned by Wikipedia.

Reply:
Looks confusing imo.

Now I think it's maybe better to remove them completely.

Although it would require duplicating the string. Turns out even merely iterating over UTF-8 `std::string` is a major PITA, as existing methods take individual bytes instead of chars.